book.query/src/main/java/com/happystays/book/query/api/queries/HotelQueryHandler.java

```diff
@@ -0,0 +1,40 @@
+package com.happystays.book.query.api.queries;
+
+import com.happystays.book.query.domain.MyTripsResponseDto;
+import com.happystays.book.query.repository.PnrRepository;
+import com.happystays.cqrs.core.domain.BaseEntity;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class HotelQueryHandler implements QueryHandler {
+    private final PnrRepository pnrRepository;
+
+    public HotelQueryHandler(PnrRepository pnrRepository) {
+        this.pnrRepository = pnrRepository;
+    }
+
+    @Override
+    public List<BaseEntity> handle(FindMyTrips query) {
+        Long id = Long.valueOf(query.getId());
+        Date d = Date.from(LocalDate.now().minusMonths(6).atStartOfDay(ZoneId.systemDefault()).toInstant());
+        var result = pnrRepository.findByUserIdAndCreationDateAfter(Long.valueOf(query.getId()), d);
+        if (result.isEmpty()) {
+            return null;
+        }
+
+        List<MyTripsResponseDto> responseDto =
+                result.stream().flatMap(pnr -> pnr.getHotelInfo().stream().flatMap(info -> info.getSegment().stream().map(seg -> new MyTripsResponseDto(seg.getConfirmationNumber(), pnr.getCreationDate(),
```

Comment:
We can format this for better understanding.
result.stream()
                        .flatMap(pnr -> pnr.getHotelInfo().stream()
                                .flatMap(info -> info.getSegment().stream()
                                        .map(seg ->
                                                new MyTripsResponseDto(seg.getConfirmationNumber(), pnr.getCreationDate(),
                                                        info.getPropertyName(), seg.getCheckInDate(), seg.getCheckOutDate(),
                                                        seg.getHotelPrice(), seg.getCurrencyCode(),
                                                        seg.getHotelCancellationInfo().getCancellationDeadline())
                                        )
                                )
                        ).sorted(Comparator.comparing(MyTripsResponseDto::getCheckInDate))
                        .collect(Collectors.toList());